src/attribute_.h

```diff
@@ -17,6 +17,70 @@ class Attribute_ : public Attribute {
     // escalated to the user.
     using Validator = std::function<std::string(T)>;
 
+    /** The constructors for Attribute_ and also for DynAttribute_ follow
+     * the a common pattern:
+     *
+     *   - The first argument is the owner object
+     *
+     *   - The second argument is the name displayed to the user
+     *
+     *   - The third argument the reading behaviour of the attribute:
+     *     * for a plain Attribute_ this is simply the initial value
+     *     * for a DynAttribute_ this is a member of the owner (or a lambda)
+     *       that returns the value of the attribute.
+     *
+     *   - The fourth argument describes the writing behaviour of the
+     *     attribute: If the fourth argument is absent, the attribute is
+     *     read-only.
+     *     * The fourth argument for an Attribute_ is a member of owner (or
+     *       lambda) that validates a new value of the attribute and returns an
+     *       error message if the new value is not acceptable for this
+     *       attribute (E.g. because a name is already taken or does not
+     *       resepct a certain format).
+     *     * The fourth argument of a DynAttribute_ is a member of owner (or
+     *       lambda) that internally processes the new value (e.g. parsing) and
+     *       returns an error message if the new value is not acceptable.
+     */
```

Comment:
Please also note the documentation on attributes describing the idea behind the constructor parameters for Attribute_ and DynAttribute_.

Reply:
The explanation was very useful to me, but in order to be a proper docstring for Doxygen, it should (afaik) be turned into a docstring for the class and/or the individual constructors. Probably a combination of both, to avoid repeating everything for each ctor.

Reply:
I think this should be attached to the header file and be included in the class description for both Attribute_ and DynAttribute_, I just don't know how to express this syntactically.